### PR TITLE
feat(probabilistic_occupancy_grid_map): add downsample filter option to ogm creation

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

This PR adds pointcloud downsample process in the occupancy grid map.

Since occupancy grid map iterates both raw and obstacle pointclouds, large number of the pointcloud input affects computation performance.
Fortunately we do not need too detailed pointcloud because grid size is often 0.5m.

Downsample topic name will be:

```
# single_lidar
/perception/occupancy_grid_map/obstacle/downsample/pointcloud
/perception/occupancy_grid_map/raw/downsample/pointcloud

# multi_lidar
/perception/occupancy_grid_map/<frame>/raw/downsample/pointcloud
/perception/occupancy_grid_map/obstacle/downsample/pointcloud
```

Sample output with tutorial rosbag:

|before|after|
|--|--|
|image|image|

Default downsample method is pickup-based filter.
Here shows downsample results in near range(~10m)
image

## Related links

Merge launcher PR first.
https://github.com/autowarefoundation/autoware_launch/pull/962

## Tests performed

Tested with Lsim with xx1-gen2.

Peformance update depends on the environment, but we can get maximum 5x faster processing time with certain situation.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by Owen-Liuyuxuan** [comment]
I roughly checked the codes. It seems that we are adding "required" parameters that may cause an error if we are using old launch files. If it is possible, I would like to see a warning or notice when we are mistakenly doing this.

Also, please add the information on the topic name in the codes' documentation.

File: perception/probabilistic_occupancy_grid_map/config/pointcloud_based_occupancy_grid_map.param.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@
       min_height: -1.0
       max_height: 2.0
 
+    # downsample input pointcloud
+    downsample_input_pointcloud: true
+    downsample_voxel_size: 0.125 # [m]
```
Comment (yukkysaito) [inline perception/probabilistic_occupancy_grid_map/config/pointcloud_based_occupancy_grid_map.param.yaml:13]: same comment
https://github.com/autowarefoundation/autoware_launch/pull/962/files#r1575513092

**Comment by yukkysaito** [comment]
@YoshiRi What will be the name of each topic when this is done?

**Comment by YoshiRi** [comment]
@Owen-Liuyuxuan 
Thanks for your check. 
> It seems that we are adding "required" parameters that may cause an error if we are using old launch files.

You are right. We need to merge https://github.com/autowarefoundation/autoware_launch/pull/962 before this.

> If it is possible, I would like to see a warning or notice when we are mistakenly doing this.

Historically, we intentionally remove default parameter deceleration so that we get error when incorrect parameter yaml is used, I think. Do you mean you do not want the node to exit when incorrect param file is used?

**Comment by YoshiRi** [comment]
@yukkysaito 
Downsampled topic will be: 

```
# single_lidar
/perception/occupancy_grid_map/obstacle/downsample/pointcloud
/perception/occupancy_grid_map/raw/downsample/pointcloud

# multi_lidar
/perception/occupancy_grid_map/<frame>/raw/downsample/pointcloud
/perception/occupancy_grid_map/obstacle/downsample/pointcloud
```

**Comment by yukkysaito** [comment]
>Downsampled topic will be:


I see :+1: Looks Good

**Comment by YoshiRi** [comment]
I add topic name description in https://github.com/autowarefoundation/autoware.universe/pull/6865/commits/bf3e7e23bed50e42c524fe701b41c9d260afef3f

**Comment by Owen-Liuyuxuan** [comment]
> I add topic name description in [bf3e7e2](https://github.com/autowarefoundation/autoware.universe/commit/bf3e7e23bed50e42c524fe701b41c9d260afef3f)

Thank you, I have no issues on my side.

**Comment by yukkysaito** [comment]
LGTM
@YoshiRi Please fix some typo (cspell) :pray:

**Comment by YoshiRi** [comment]
@yukkysaito 
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/6865/commits/594c3bba86db31ce7863f8ebbb572500b87244eb.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**